compiler/x/amd64/codegen/OMRMemoryReference.cpp

```diff
@@ -392,7 +392,7 @@ OMR::X86::AMD64::MemoryReference::addMetaDataForCodeAddressWithLoad(
             {
             if (cg->needClassAndMethodPointerRelocations())
                cg->addAOTRelocation(new (cg->trHeapMemory()) TR::ExternalRelocation(displacementLocation, (uint8_t *)&_symbolReference,
-                                                                                        (uint8_t *)containingInstruction->getNode()->getInlinedSiteIndex(),
+                                                                                        (uint8_t *)(uintptr_t)containingInstruction->getNode()->getInlinedSiteIndex(),
```

Comment:
We should try to prefer `static_cast` over C-style casts for additional compiler provided safety. It catches obvious bugs and makes refactoring easy. For example changing the type `containingInstruction` returns to something invalid will result in a compile time bug rather than runtime crash.

Also `getInlinedSiteIndex` seems to return an `int16_t`. Why are these casts necessary?

https://github.com/eclipse/omr/blob/3cda7d85e70ba406301453269cd578f76f68af1e/compiler/il/OMRNode.hpp#L700

Reply:
You get this warning without it `warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]`.  You first have to cast it to an integer of the correct size and then you can cast it to a pointer

Reply:
We need to clean this up. I just looked at uses of `TR::ExternalRelocation::_targetAddress2` (bad name IMO) and it seems it is being used in all sorts of contexts. If the respective field is to be used as an arbitrary cast it should be of type `void*` to make this explicit.

To avoid all the casting at source we should provide `int16_t` constructors and cast to `void*` underneath the hood if need be.

Reply:
https://github.com/eclipse/omr/issues/2237